src/nvim/normal.c

```diff
@@ -1657,8 +1672,11 @@ void do_pending_operator(cmdarg_T *cap, int old_col, bool gui_yank)
       if (empty_region_error) {
         vim_beep();
         CancelRedo();
-      } else
-        op_replace(oap, cap->nchar);
+      } else {
+          // Restore linebreak, so that when the user edits it looks as before.
+          curwin->w_p_lbr = lbr_saved;
+          op_replace(oap, cap->nchar);
```

Comment:
Indent with two spaces. (Have you seen our style guide? http://neovim.io/develop/style-guide.xml)